### PR TITLE
ckanextensions

### PR DESCRIPTION
**Pull request by fdelavega**
Include documentation in CKAN extensions and monetization framework

### CODE REVIEW

File: doc/publishing-open-data-in-fiware/fiware-extended-ckan.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# CKAN Monetization Architecture
+
+In the context of the FIWARE platform, CKAN is used as the catalog where
+data can be published, discovered, managed, and consumed. Within the FIWARE
+platform, not only static datasets (CSV, XLSX, etc) are supported, but also
+right time context information can be published in CKAN as Context Broker
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/fiware-extended-ckan.md]: real (right)

File: doc/publishing-open-data-in-fiware/fiware-extended-ckan.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+queries (pollutants level in Madrid, etc.)
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/fiware-extended-ckan.md]: for instance, pollutants level in Madrid, etc.)

File: doc/publishing-open-data-in-fiware/fiware-extended-ckan.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+In addition, FIWARE supports to manage the access and monetize the published
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/fiware-extended-ckan.md]: In addition, using the Biz Ecosystem GE, FIWARE provides the components that allow to manage the access and monetisation of published datasets.

File: doc/publishing-open-data-in-fiware/fiware-extended-ckan.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+[<img src="../images/ckan-monetization-arch.png" srcset="../images/ckan-monetization-arch.png 2x" alt="CKAN Monetization Architecture"/>](images/ckan-monetization-arch.png)
+
+As can be seen in the previous image, the following components are
+integrated in order to manage and monetize data:
+
+- Extended CKAN: Used for the publication of the static datasets and
+the Context Broker queries.
+- Biz Ecosystem: Used for the creation and acquisition of data offerings,
+including customer charging, and access grant
+- IdM + PEP + PDP (Security Framework): Used for provision of identities,
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/fiware-extended-ckan.md]: used for the provision

File: doc/publishing-open-data-in-fiware/fiware-extended-ckan.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+- For static datasets: The Biz Ecosystem uses the API exposed by the
+private datasets CKAN extension (described later) in order to include
+the user in the access list of the particular dataset.
+- For Context Broker queries: The Biz Ecosystem also grants access to the
+context broker service by giving a particular role (defined by the provider)
+to the customer user. This role has a bound policy that specifies the
+permissions to access to the acquired query.
+
+To support the aforementioned functionality, it has been needed to
+extend CKAN with features as NGSI management, OAUth2, or datasets privacy
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/fiware-extended-ckan.md]: as NGSI support, OAuth2

File: doc/publishing-open-data-in-fiware/fiware-extended-ckan.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+This extension provides an HTTP API that allows to add and remove users from
+the dataset access list, enabling the Biz Ecosystem to automatically 
+manage these permissions.
+
+This extension is available in [GitHub](https://github.com/conwetlab/ckanext-privatedatasets)
+
+
+## NGSI View
+
+This CKAN extension provides the ability to generate right-time resources
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/fiware-extended-ckan.md]: real time (or right time)

File: doc/publishing-open-data-in-fiware/how-to-publish-open-datasets-in-ckan-2.md [hunk 7]
```diff
@@ -23,23 +23,49 @@ the visibility of the dataset:
...
-​​[![HowToPublishDatasheetsInCkan2](images/HowToPublishDatasheetsInCkan21.png)](images/HowToPublishDatasheetsInCkan21.png)
+[![HowToPublishDatasheetsInCkan2](images/HowToPublishDatasheetsInCkan21.png)](images/HowToPublishDatasheetsInCkan21.png)
 
 In the second step you have to provide the data itself. To do that, you can
 provide a link or upload a file with the data. Any type of file is allowed, but if you
 want to generate an automatic API to access your data, you must upload a
 CSV file.
 
+> **NOTE**: Take into account that “Private” resources created using the link
+> option must be secured using the appropiated mechanism (e.g. by following the
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/how-to-publish-open-datasets-in-ckan-2.md]: appropriate

File: doc/publishing-open-data-in-fiware/how-to-publish-open-datasets-in-ckan-2.md [hunk 8]
```diff
@@ -23,23 +23,49 @@ the visibility of the dataset:
...
+
 [![HowToPublishDatasheetsInCkan3](images/HowToPublishDatasheetsInCkan3.png)](images/HowToPublishDatasheetsInCkan3.png)
 
+You can also provide dynamic context information by providing a context broker
+URL and selecting the `fiware-ngsi` format. Once selected the `fiware-ngsi`
+format, three extra fields will appear allowing you to configure the tenant and
+the service path to use for accesing the context broker (you can leave them
+empty if the data is stored in the default tenant or service path). The last
+field allows you to indicate whether the context broker requires authentication
+(see the [Handling authorization and access control to APIs][]
```
Comment (jmcanterafonseca) [inline doc/publishing-open-data-in-fiware/how-to-publish-open-datasets-in-ckan-2.md]: is this link properly set?

Comment (fdelavega) [inline doc/publishing-open-data-in-fiware/how-to-publish-open-datasets-in-ckan-2.md]: Yes, there is a tag at the end of the file

    [Handling authorization and access control to APIs]: /handling-authorization-and-access-control-to-apis/introduction/

**Comment by aarranz** [comment]
@jmcanterafonseca please, merge it ASAP as Juanjo want to use those contents in a meeting.

**Pull request merged**